presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -212,6 +216,35 @@ public SubPlan createSubPlans(Session session, Plan plan, boolean forceSingleNod
         return subPlan;
     }
 
+    /**
+     * expected query pattern to be executed on coordinator:
+     * `SHOW CREATE TABLE table_name`,`EXPLAIN statement`,`DESCRIBE INPUT[OUTPUT] statement_name`,`SHOW STATS FOR table`: OutputNode > ValuesNode
+     * `select 1`: OutputNode > ProjectNode > ExchangeNode > ValuesNode
+     *
+     * @param planNode
+     * @return
+     */
+    private boolean canForceCoordinatorExecution(PlanNode planNode)
```

Comment:
I would recommend implementing this check as a `PlanVisitor`: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/sql/planner/SimplePlanVisitor.java. As long as the plan contains only `ValuesNode` in it's leaf node the query can be executed locally.

Reply:
finish